src/dune_engine/build_system.mli

```diff
@@ -122,7 +122,13 @@ module Error : sig
 
   val create : exn:Exn_with_backtrace.t -> t
 
-  val info : t -> User_message.t * User_message.t list * Path.t option
+  type info =
```

Comment:
I find `main` and `related` not sufficiently self-describing. Could you add some docs here?